### PR TITLE
Migrating hasUniqueTag to Crane

### PR DESCRIPTION
**Pull request by yashoza19**

### CODE REVIEW

File: certification/internal/policy/container/skopeo.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package container
```
Comment (bcrochet) [inline certification/internal/policy/container/skopeo.go:1]: This file should go into certification/internal/engine

File: certification/engine/engine.go [hunk 2]
```diff
@@ -164,21 +165,22 @@ var operatorPolicy = map[string]certification.Check{
 }
 
 var containerPolicy = map[string]certification.Check{
-	hasLicenseCheck.Name():        hasLicenseCheck,
-	maxLayersCheck.Name():         maxLayersCheck,
-	hasLicenseCheck.Name():        hasLicenseCheck,
-	hasNoProhibitedCheck.Name():   hasNoProhibitedCheck,
-	hasRequiredLabelsCheck.Name(): hasRequiredLabelsCheck,
+	hasLicenseCheck.Name():        	hasLicenseCheck,
+	hasUniqueTagCheck.Name():		hasUniqueTagCheck,
```
Comment (bcrochet) [inline certification/engine/engine.go:169]: Formatting looks off here.

File: certification/engine/engine.go [hunk 3]
```diff
@@ -164,21 +165,22 @@ var operatorPolicy = map[string]certification.Check{
...
-	basedOnUbiCheck.Name():                 basedOnUbiCheck,
-	deprecatedHasLicenseCheck.Name():       deprecatedHasLicenseCheck,
-	hasUniqueTagCheck.Name():               hasUniqueTagCheck,
-	deprecatedHasNoProhibitedCheck.Name():  deprecatedHasNoProhibitedCheck,
+	runAsNonRootCheck.Name():               	runAsNonRootCheck,
+	deprecatedUnderLayerMaxCheck.Name():    	deprecatedUnderLayerMaxCheck,
+	deprecatedHasRequiredLabelCheck.Name(): 	deprecatedHasRequiredLabelCheck,
+	basedOnUbiCheck.Name():                 	basedOnUbiCheck,
+	deprecatedHasLicenseCheck.Name():       	deprecatedHasLicenseCheck,
+	deprecatedHasUniqueTagCheck.Name():			deprecatedHasUniqueTagCheck,
```
Comment (bcrochet) [inline certification/engine/engine.go:182]: Formatting looks off here. Run 'make fmt'.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Couple of comments to start. Make sure you look at the Validate Operator Bundle check to see how to inject the SkopeoEngine.

File: certification/internal/policy/container/has_unique_tag.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	if err != nil {
+		log.Error("unable to execute skopeo on the image: ", err)
+		return nil, err
+	}
+
+	return runReport.Tags, nil
+}
+
+func (p *HasUniqueTagCheck) validate(image string, tags []string) (bool, error) {
```
Comment (samira-barouti) [inline certification/internal/policy/container/has_unique_tag.go:44]: It seems that the `image` arg is not being used. Could you please remove it?

**Review by samira-barouti — CHANGES_REQUESTED** [review]

File: certification/internal/engine/skopeo.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+func NewSkopeoEngine() *cli.SkopeoEngine {
+	var engine cli.SkopeoEngine = SkopeoEngine{}
+	return &engine
+}
+
+type SkopeoEngine struct{}
+
+func (e SkopeoEngine) ListTags(image string) (*cli.SkopeoListTagsReport, error) {
+
+	imageName, err := e.imageName(image)
```
Comment (sebrandon1) [inline certification/internal/engine/skopeo.go:24]: Does the call to `imageName` (and its subsequent functions) need to be part of the `SkopeoEngine` struct?  It seems that they don't use the struct's vars or funcs after the initial call to `imageName`.

**Review by sebrandon1 — COMMENTED** [review]
One comment about the `imageName` funcs.

File: certification/internal/engine/skopeo.go (hunk 5 above)
Comment (samira-barouti) [reply certification/internal/engine/skopeo.go:24]: This method is used to strip digest/sha from the image name. Skopeo is the only engine that needs this functionality; but `imageName` method can be moved to util/container for the future.

Comment (sebrandon1) [reply certification/internal/engine/skopeo.go:24]: Thanks for clarifying!

**Review by samira-barouti — APPROVED** [review]
/lgtm

**Review by bcrochet — APPROVED** [review]

**Review by bcrochet — APPROVED** [review]

**Pull request merged**